jablib/src/main/java/org/jabref/logic/formatter/bibtexfields/ConvertMSCCodesFormatter.java

```diff
@@ -0,0 +1,89 @@
+package org.jabref.logic.formatter.bibtexfields;
+
+import java.net.URL;
+import java.util.Objects;
+
+import org.jabref.logic.cleanup.Formatter;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.layout.LayoutFormatter;
+import org.jabref.logic.shared.exception.MscCodeLoadingException;
+import org.jabref.logic.util.MscCodeUtils;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConvertMSCCodesFormatter extends Formatter implements LayoutFormatter {
+    private static final Logger logger = LoggerFactory.getLogger(ConvertMSCCodesFormatter.class);
+    private static final BiMap<String, String> MSCMAP;
+    private static boolean conversionPossible;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConvertMSCCodesFormatter.class);
+
+    static {
+
+        HashBiMap<String, String> tempMap = HashBiMap.create();
+
+        URL resourceUrl = ConvertMSCCodesFormatter.class.getClassLoader().getResource("msc_codes.json");
+
+        // Check for valid mapping of msc codes
+        if (resourceUrl == null) {
+            logger.error("Resource not found: msc_codes.json");
+            conversionPossible = false;
+        } else {
+            try {
+                tempMap = MscCodeUtils.loadMscCodesFromJson(resourceUrl).get();
+                if (!tempMap.isEmpty()) {
+                    conversionPossible = true;
+                }
+            } catch (MscCodeLoadingException e) {
+                logger.error("Error loading MSC codes:", e);
+                conversionPossible = false;
+            }
+        }
+
+        MSCMAP = tempMap;
+    }
+
+    @Override
+    public String format(String text) {
+        StringBuilder result = new StringBuilder(Objects.requireNonNull(text));
+
+        if (result.isEmpty() || !conversionPossible) {
+            return result.toString();
+        }
+        // text contains comma separated codes
+        String[] codeList = result.toString().split(",");
```

Comment:
Use `org.jabref.model.entry.KeywordList#parse(java.lang.String, java.lang.Character, java.lang.Character)` and then work on `KeywordList`

Reason: Better data structure and adheres to user configuration (keyword delimiter) and escpaings.

See at other usages of the parser and how to pass the prefernces

---

At first, I thought:

```suggestion
        String[] codeList = text.toString().split(",");
```

but this is wrong, because no escaping treatment etc.